src/main/java/com/exasol/containers/VersionBasedExasolDockerImageReference.java

```diff
@@ -91,12 +81,29 @@ public boolean hasDockerImageRevision() {
         return this.dockerImageRevision != VERSION_NOT_PRESENT;
     }
 
+    @Override
+    public String getSuffix() {
+        return this.suffix;
+    }
+
+    @Override
+    public boolean hasSuffix() {
+        return this.suffix != null;
+    }
+
     @Override
     public String toString() {
-        if (!this.hasDockerImageRevision() && (this.major >= 7)) {
-            return EXASOL_DOCKER_IMAGE_ID + ":" + this.major + "." + this.minor + "." + this.fix;
+        if (!this.hasDockerImageRevision() && isExasolSevenOrLater()) {
+            return EXASOL_DOCKER_IMAGE_ID + ":" + this.major + "." + this.minor + "." + this.fix
+                    + (this.suffix == null ? "" : "-" + this.suffix);
+        } else {
+            return EXASOL_DOCKER_IMAGE_ID + ":" + this.major + "." + this.minor + "." + this.fix
+                    + (this.suffix == null ? "" : "-" + this.suffix) + "-d"
+                    + (this.hasDockerImageRevision() ? this.dockerImageRevision : 1);
```

Comment:
```suggestion
       return EXASOL_DOCKER_IMAGE_ID + ":" + this.major + "." + this.minor + "." + this.fix
                    + (this.suffix == null ? "" : "-" + this.suffix) + getDockerImageRevision();
                    ....
       private String getDockerImageRevision() {
           return ((!this.hasDockerImageRevision() && isExasolSevenOrLater()) ? "-d"
                    + (this.hasDockerImageRevision() ? this.dockerImageRevision : 1) : null;
        }
        
```

Also check the if, seems a bit strange that older releases always run into the else branch!

Reply:
Made the code more readable by extracting the construction of the reference parts and moving the version distinction down to the image revision construction, where it now has a localized impact.